src/sysupdate/sysupdate-transfer.c

```diff
@@ -894,6 +1062,13 @@ int transfer_acquire_instance(Transfer *t, Instance *i) {
                         return log_oom();
         }
 
+        struct notify_userdata notify_ctx = {
+                .transfer = t,
+                .instance = i,
+                .callback = cb,
+                .userdata = userdata,
+        };
```

Comment:
i think this should really be initialized as part of run_callout() above, just pass t + i to it, so that it can fill this in on its own